phoenix-core/src/main/java/org/apache/phoenix/jdbc/LoggingPhoenixStatement.java

```diff
@@ -59,4 +70,17 @@ public ResultSet getGeneratedKeys() throws SQLException {
         return new LoggingPhoenixResultSet(super.getGeneratedKeys(), phoenixMetricsLog, this.sql);
     }
 
+    private void loggingAutoCommitHelper() throws SQLException {
+        Connection conn = this.getConnection();
+
+        if(stmt.getUpdateOperation().isMutation() && conn.getAutoCommit()){
```

Comment:
Isn't `conn.getAutoCommit()` check sufficient?

Reply:
Follow up: Is `Statement stmt` reference needed?

Reply:
This helper function gets called from both execute and executeUpdate. We need Statment reference to distinguish in case of execute

Reply:
What happens if its not a mutation and we try logging the metrics (in case of queries)?

Reply:
I see this piece of code
```
                        // If transactional, this will move the read pointer forward
                        if (connection.getAutoCommit()) {
                            connection.commit();
                        }
```
@twdsilva any idea?

Reply:
Yes, in case of executeQuery, metrics get logged. The differentiating function call happens in PhoenixStatement Class. In case of mutation, it directly executes executeMutation.

Reply:
Not sure if that is needed any more since we start the transaction in MetaDataClient.updateCache

boolean isTransactional = (table!=null && table.isTransactional());
        if (isTransactional) {
            connection.getMutationState().startTransaction(table.getTransactionProvider());
        }

Reply:
> In case of mutation, it directly executes executeMutation

You can also provide a mutation through `execute()` method call.